### PR TITLE
RFE: Add basic support for luks and luks-inside-qcow2 in avocado-vt

### PR DESCRIPTION
**Pull request by hereischen**
id: 1528107
Signed-off-by: Haotong Chen hachen@redhat.com

### CODE REVIEW

File: shared/cfg/guest-hw.cfg [hunk 1]
```diff
@@ -114,6 +114,9 @@ variants:
         # placeholder
 
 variants:
+    - luks:
+        image_format = luks
```
Comment (xutian) [inline shared/cfg/guest-hw.cfg]: Hi guy
it looks that LUKS is a disk encryption specification, not a image format, right?

File: virttest/qemu_devices/qcontainer.py [hunk 2]
```diff
@@ -1584,6 +1600,14 @@ def images_define_by_params(self, name, image_params, media=None,
                 scsi_hba = "virtio-scsi-device"
             elif "s390" in image_params.get("machine_type"):
                 scsi_hba = "virtio-scsi-ccw"
+        # support for luks for .luks and .qcow2
+        # .qcow2 requires "encryption = luks"
+        encryption = image_params.get("encryption")
+        img_cipher = image_params.get("image_cipher")
+        img_fmt = image_params.get("image_format")
+        if img_fmt == "luks":
```
Comment (xutian) [inline virttest/qemu_devices/qcontainer.py]: I think it's not a image format here.

File: virttest/qemu_storage.py [hunk 3]
```diff
@@ -90,51 +125,58 @@ def create(self, params, ignore_errors=False):
...
+                    _extend_option(cmd, "-o", [("encrypt.format", "luks"),
+                                               ("encrypt.key-secret", sec_id)])
+                # TODO: base + luks encryption
+                # for now you can ether use luks or base, not both
+                if self.base_tag:
+                    _add_option(cmd, "-b", [self.base_image_filename])
+                    if self.base_format:
+                        _add_option(cmd, "-F", [self.base_format])
+
+            if self.image_format == "luks":
```
Comment (xutian) [inline virttest/qemu_storage.py]: I think here should be
if  encryption  == "luks":
     .....

File: shared/cfg/guest-hw.cfg (hunk 1 above)
Comment (hereischen) [reply shared/cfg/guest-hw.cfg]: No, there is .luks image format.
Please refer to
https://qemu.weilnetz.de/doc/qemu-doc.html#disk_005fimages_005fformats

eg:
>     #qemu-img create -f luks --object secret,id=sec0,data=`printf %s
> "redhat" |base64`,format=base64 -o key-secret=sec0 demo.luks 5G
>     #qemu-img info demo.luks 
>     image: demo.luks
>     file format: luks
>     virtual size: 5.0G (5368709120 bytes)
>     disk size: 256K
>     encrypted: yes

**Comment by hereischen** [comment]
This is a work around for luks, please don't review it.

**Comment by XueqiangWei** [comment]
tested with virti-scsi and rhel7.6, hotplug related cases are failed.

**Comment by hereischen** [comment]
That is expected, as the patch does not support hot-plug.

**Pull request closed without merging**